lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -843,6 +843,22 @@ bool SwiftLanguageRuntimeImpl::GetObjectDescription(Stream &str,
   return RunObjectDescriptionExpr(object, expr_string, str);
 }
 
+StructuredDataImpl *
+SwiftLanguageRuntime::GetLanguageSpecificData(StackFrame &frame) {
+  auto sc = frame.GetSymbolContext(eSymbolContextFunction);
```

Comment:
I think the use of auto in this function is hurting rather than helping the readability and diverges from the [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable).

Reply:
> or other places where the type is already obvious from the context

seems this is not very helpful guidance, since it doesn't give examples of when this is true and when it's not true.

I am probably guilty, in some eyes, of using auto too much, but I do apply some judgment. In this case:

```cpp
auto sc = frame.GetSymbolContext(eSymbolContextFunction);
```

^ I felt that `GetSymbolContext` made the type obvious.

```cpp
auto dict_sp = std::make_shared<StructuredData::Dictionary>();
```

^ I felt `make_shared<T>` made the type obvious, also the `_sp` suffix.

```cpp
auto symbol = sc.function->GetMangled().GetMangledName().GetStringRef();
```

^ I felt `GetStringRef` made the type obvious. In hindsight, having the type be on the far right of a chain of method calls has downsides.

```cpp
auto is_async = SwiftLanguageRuntime::IsAnySwiftAsyncFunctionSymbol(symbol);
```

^ I felt the `is_` prefix was a good indication, but in hindsight this is better with `bool`.

```cpp
auto *data = new StructuredDataImpl;
```

^ I felt `new T` made the type.

Reply:
> From that link:
> 
> > or other places where the type is already obvious from the context
> 
> It seems this is not very helpful guidance, since it doesn't give examples of when this is true, and when it's not true.

I agree, but I suspect that's on purpose to allow some freedom/discretion based on the situation. 

> I am probably guilty, in some eyes, of using auto too much, but I do apply some judgment. 

I think we must have a different definition of obvious :-) I don't want to go over them one-by-one, but if I were to play the devil's advocate I think I could make an argument against all of them, except the ones that actually mention the types (as types) in the RHS like the call to `new` and `make_shared`. Anyway, not a big deal, but I think the way it's used here is a stretch compared to the rest of LLVM.

Reply:
So one thing about auto that bugs me is that most IDE's have a "jump to definition" feature.  So for instance, if I'm new to lldb and see:

SymbolContext *sym_ctx = something.GetSymbolContext();

and I want to know what a SymbolContext is, I can command click on the text and I'm there.  But if I see:

auto *sym_ctx = something.GetSymbolContext();

I have to command-click on GetSymbolContext, find the return type of THAT, and then click on that.  So even these seemingly innocent uses of auto do make it harder to get into the code...

Reply:
True, it's tricky. Some IDEs let you command click on the variable name to navigate to its type, but not all. Another option is to hover on the function, and then from the popover window, click its return type, but that also doesn't always work. I'll use shift to using more named types.